### PR TITLE
Blog app - integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by MrOmachi**
**In this pull request I have implemented the following**

- Write integration tests with Capybara gem.
- Secure app from n+1 problems.

### CODE REVIEW

File: spec/views/user_view_spec.rb [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    it 'Not to show the Pagination button' do
+      expect(page.body).not_to include('Pagination')
+    end
+
+    it 'Redirect to post show page when a post is clicked' do
+      click_link(@last_post.title)
+      expect(current_path).to eq(user_post_path(@user, @last_post))
+    end
+  end
+end
```
Comment (GzimAsani) [inline spec/views/user_view_spec.rb]: - [ ] User index and show page should be tested on different spec.rb files

Comment (GzimAsani) [reply spec/views/user_view_spec.rb]: - [ ] Please add these tests for users index page 
image

File: spec/views/user_view_spec.rb [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+                            likes_counter: 0)
+    Comment.create(post: @any_post, author_id: @user.id, text: 'This is the 1st comment on the 4th post')
+    Comment.create(post: @any_post, author_id: @user.id, text: 'This is the 2nd comment on the 4th post')
+    Comment.create(post: @any_post, author_id: @user.id, text: 'This is the 3rd comment on the 4th post')
+    visit user_post_path(@user, @any_post)
+  end
+
+  it 'shows number of likes a post has' do
+    expect(page.body).to include('Likes: 0')
+  end
```
Comment (GzimAsani) [inline spec/views/user_view_spec.rb]: - [x] For the user show page please add a test that asserts you can see the number of posts the user has written.

Comment (GzimAsani) [reply spec/views/user_view_spec.rb]: - [x] For users show page please add a test that asserts you can see the user's first 3 posts.
*

Comment (GzimAsani) [reply spec/views/user_view_spec.rb]: for user show page please add a test that asserts you can see a button that lets me view all of a user's posts.

Comment (GzimAsani) [reply spec/views/user_view_spec.rb]: For user show page please add a test that redirects to the user's post's index page when you click on a **see all posts**.

File: spec/views/post_view_spec.rb [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    it 'Not to show the Pagination button' do
+      expect(page.body).not_to include('Pagination')
+    end
+
+    it 'Redirect to post show page when a post is clicked' do
+      click_link(@last_post.title)
+      expect(current_path).to eq(user_post_path(@user, @last_post))
+    end
+  end
+end
```
Comment (GzimAsani) [inline spec/views/post_view_spec.rb]: - [ ] Post index and show page should be on different file, please add the post index page and include the following tests
image

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello  @MrOmachi                       :wave:   
 Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good with testing, you made the majority of them:+1: 

 ## Required Changes 
- [ ] On your post controller index, fix the n+1 problem for fetching the posts together with comments by doing something similar like this:
@posts = @user.posts.includes(:comments)
- [ ] Check the comments below :arrow_double_down: 




 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by MrOmachi** [comment]
Thanks for the review, All reviews have been implemented

**Review by SajjadAhmad14 — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

### Hi Mr Omachi, 
## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
Git fow is followed. :heavy_check_mark:
## Congratulations 🎉 Thank you for implementing all the requirements 🙏
### Please go ahead and merge the branch!

## Happy Coding 👏👏👏
#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**